src/main/java/com/onehouse/metadata_extractor/ActiveTimelineInstantBatcher.java

```diff
@@ -128,30 +170,64 @@ public List<List<File>> createBatches(List<File> instants, int maxBatchSize) {
       batches.add(currentBatch);
     }
 
-    return batches;
+    return Pair.of(checkpoint, batches);
+  }
+
+  private static String decrementNumericString(String numericString) {
+    // Convert the string to a BigInteger
+    BigInteger number = new BigInteger(numericString);
+
+    // Decrement the value by 1
+    BigInteger decrementedNumber = number.subtract(BigInteger.ONE);
```

Comment:
the timestamps in the instant files are not guranteed to be 1ms appart. decrementing by 1 is not reliable